src/pubtools/sign/signers/cosignsigner.py

```diff
@@ -0,0 +1,291 @@
+from __future__ import annotations
+
+from dataclasses import field, dataclass
+import enum
+import json
+import logging
+from typing import Dict, List, ClassVar, Any
+import os
+import sys
+
+import click
+
+from . import Signer
+from ..operations.base import SignOperation
+from ..operations import ContainerSignOperation
+from ..results.signing_results import SigningResults
+from ..results import ContainerSignResult
+from ..results import SignerResults
+from ..exceptions import UnsupportedOperation
+from ..conf.conf import load_config, CONFIG_PATHS
+from ..utils import set_log_level, run_command, _get_config_file
+
+
+LOG = logging.getLogger("pubtools.sign.signers.cosignsigner")
+
+
+class SignRequestType(str, enum.Enum):
+    """Sign request type enum."""
+
+    CONTAINER = "container_signature"
+    CLEARSIGN = "clearsign_signature"
+
+
+@dataclass()
+class CosignSignerResults(SignerResults):
+    """MsgSignerResults model."""
+
+    status: str
+    error_message: str
+
+    def to_dict(self: SignerResults):
+        """Return dict representation of MsgSignerResults model."""
+        return {"status": self.status, "error_message": self.error_message}
+
+    @classmethod
+    def doc_arguments(cls: SignerResults) -> Dict[str, Any]:
+        """Return dictionary with result description of SignerResults."""
+        doc_arguments = {
+            "signer_result": {
+                "type": "dict",
+                "description": "Signing result status.",
+                "returned": "always",
+                "sample": {"status": "ok", "error_message": ""},
+            }
+        }
+
+        return doc_arguments
+
+
+@dataclass()
+class CosignSigner(Signer):
+    """Messaging signer class."""
+
+    cosign_bin: str = field(
+        init=False,
+        metadata={
+            "description": "Path to cosign binary",
+            "sample": "/usr/local/bin/cosign",
+        },
+        default="/usr/bin/cosign",
+    )
+    timeout: str = field(
+        init=False,
+        metadata={
+            "description": "Timeout for cosign operations with units",
+            "sample": "60s",
+        },
+        default="3m0s",
+    )
+    allow_http_registry: bool = field(
+        init=False,
+        metadata={
+            "description": "Allow http registry",
+            "sample": False,
+        },
+        default=False,
+    )
+    allow_insecure_registry: bool = field(
+        init=False,
+        metadata={
+            "description": "Allow insecure registry",
+            "sample": False,
+        },
+        default=False,
+    )
+    rekor_url: str = field(
+        init=False,
+        metadata={
+            "description": "URL for rekor stl server",
+            "sample": "https://rekor.sigstore.dev",
+        },
+        default="https://rekor.sigstore.dev",
```

Comment:
no, it's actually bad enough there are even some defaults. You can easily rely on some default values and when they change them you might not even notice that. Therefore I rather copied them so we can control if these are changed or not

Reply:
I assume if the default rekor URL changes, we will want to use the new one. Do you foresee a scenario where that's not the case?

Reply:
Then we change it as well. For rekor url it probably doesn't make sense to have default, but for timeout for example, when they change it to double it can be unpleasant surprise for us.